CI failing on `main`

I'm seeing lots of failures on `main`. See, for example, [this CI build](https://github.com/dask/dask/actions/runs/8186791667/job/22385899246). Here are a couple of example failures:

```python
_____________________ test_groupby_shift_lazy_input[tasks] _____________________
[gw0] linux -- Python 3.12.2 /home/runner/miniconda3/envs/test-environment/bin/python3.12

    @pytest.mark.xfail(DASK_EXPR_ENABLED, reason="delayed not currently supported in here")
    def test_groupby_shift_lazy_input():
        pdf = pd.DataFrame(
            {
                "a": [0, 0, 1, 1, 2, 2, 3, 3, 3],
                "b": [4, 5, 6, 3, 2, 1, 0, 0, 0],
                "c": [0, 0, 0, 0, 0, 1, 1, 1, 1],
            },
        )
        delayed_periods = dask.delayed(lambda: 1)()
        ddf = dd.from_pandas(pdf, npartitions=3)
>       assert_eq(
            pdf.groupby(pdf.c).shift(periods=1),
            ddf.groupby(ddf.c).shift(periods=delayed_periods, meta={"a": int, "b": int}),
        )

dask/dataframe/tests/test_groupby.py:2631: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
dask/dataframe/utils.py:585: in assert_eq
    b = _check_dask(
dask/dataframe/utils.py:474: in _check_dask
    result = dsk.compute(scheduler=scheduler)
../../../miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/_collection.py:421: in compute
    return DaskMethodsMixin.compute(out, **kwargs)
dask/base.py:375: in compute
    (result,) = compute(self, traverse=False, **kwargs)
dask/base.py:661: in compute
    results = schedule(dsk, keys, **kwargs)
../../../miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/_expr.py:3553: in _execute_task
    return dask.core.get(graph, name)
../../../miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/_groupby.py:1138: in operation
    return dask_func(
../../../miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/_groupby.py:1198: in groupby_slice_shift
    return _groupby_slice_shift(
dask/dataframe/groupby.py:264: in _groupby_slice_shift
    result = g.shift(**kwargs)
lib.pyx:1241: in pandas._libs.lib.is_list_like
    ???
lib.pyx:1259: in pandas._libs.lib.c_is_list_like
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Delayed('eq-6925980ea13ac5c178bf40e999ab203b')

    def __bool__(self):
>       raise TypeError("Truth of Delayed objects is not supported")
E       TypeError: Truth of Delayed objects is not supported

dask/delayed.py:645: TypeError
```

```python
_________ test_dataframe_aggregations_multilevel[cov-disk-1-<lambda>3] _________
[gw3] linux -- Python 3.12.2 /home/runner/miniconda3/envs/test-environment/bin/python3.12

grouper = <function <lambda> at 0x7f45d4d27560>, agg_func = 'cov', split_out = 1

    @pytest.mark.flaky(reruns=5)  # See https://github.com/dask/dask/issues/9793
    @pytest.mark.parametrize(
        "grouper",
        [
            lambda df: ["a"],
            lambda df: ["a", "b"],
            lambda df: df["a"],
            lambda df: [df["a"], df["b"]],
            lambda df: [df["a"] > 2, df["b"] > 1],
        ],
    )
    @pytest.mark.parametrize("split_out", [1, 2])
    def test_dataframe_aggregations_multilevel(grouper, agg_func, split_out):
        sort = split_out == 1  # Don't sort for split_out > 1
    
        def call(g, m, **kwargs):
            return getattr(g, m)(**kwargs)
    
        pdf = pd.DataFrame(
            {
                "a": [1, 2, 6, 4, 4, 6, 4, 3, 7] * 10,
                "b": [4, 2, 7, 3, 3, 1, 1, 1, 2] * 10,
                "d": [0, 1, 2, 3, 4, 5, 6, 7, 8] * 10,
                "c": [0, 1, 2, 3, 4, 5, 6, 7, 8] * 10,
            },
            columns=["c", "b", "a", "d"],
        )
    
        ddf = dd.from_pandas(pdf, npartitions=10)
    
        # covariance only works with N+1 columns
        if agg_func not in ("cov", "corr"):
            assert_eq(
                call(pdf.groupby(grouper(pdf), sort=sort)["c"], agg_func),
                call(
                    ddf.groupby(grouper(ddf), sort=sort)["c"],
                    agg_func,
                    split_out=split_out,
                    split_every=2,
                ),
            )
    
        # not supported by pandas
        if agg_func != "nunique":
            if agg_func in ("cov", "corr") and split_out > 1:
                pytest.skip("https://github.com/dask/dask/issues/9509")
>           assert_eq(
                call(pdf.groupby(grouper(pdf), sort=sort)[["c", "d"]], agg_func),
                call(
                    ddf.groupby(grouper(ddf), sort=sort)[["c", "d"]],
                    agg_func,
                    split_out=split_out,
                    split_every=2,
                ),
            )

dask/dataframe/tests/test_groupby.py:1393: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
dask/dataframe/utils.py:585: in assert_eq
    b = _check_dask(
dask/dataframe/utils.py:474: in _check_dask
    result = dsk.compute(scheduler=scheduler)
../../../miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/_collection.py:421: in compute
    return DaskMethodsMixin.compute(out, **kwargs)
dask/base.py:375: in compute
    (result,) = compute(self, traverse=False, **kwargs)
dask/base.py:661: in compute
    results = schedule(dsk, keys, **kwargs)
../../../miniconda3/envs/test-environment/lib/python3.12/site-packages/dask_expr/_groupby.py:640: in aggregate
    return _cov_agg(inputs[0], **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

_t = [[[[(     0  3
2 1      
1 4  0  0
2 2  1  1
3 1  7  7
4 1  6  6
  3  7  7
6 1  5  5
  7  2  2
7 2  8  8,      00  03 ...     1        1
  7        1        1
7 2        1        1, OrderedDict({'c': '0', 'b': '1', 'a': '2', 'd': '3'}))]]]]
levels = [0, 1], ddof = 1, std = False, sort = True

    def _cov_agg(_t, levels, ddof, std=False, sort=False):
        sums = []
        muls = []
        counts = []
    
        # sometime we get a series back from concat combiner
        t = list(_t)
    
>       cols = t[0][0].columns
E       AttributeError: 'list' object has no attribute 'columns'

dask/dataframe/groupby.py:692: AttributeError
```

cc @phofl @hendrikmakait @fjetter for the `dask-expr` connection

--- a/dask/dataframe/io/tests/test_io.py
+++ b/dask/dataframe/io/tests/test_io.py
@@ -835,7 +835,6 @@ def test_to_delayed():
     assert_eq(dx.compute(), x)
 
 
-@pytest.mark.xfail(DASK_EXPR_ENABLED, reason="doesn't make sense at the moment")
 def test_to_delayed_optimize_graph():
     df = pd.DataFrame({"x": list(range(20))})
     ddf = dd.from_pandas(df, npartitions=20)

